sycl/source/detail/device_impl.hpp

```diff
@@ -1195,6 +1195,15 @@ class device_impl : public std::enable_shared_from_this<device_impl> {
             "The device does not have the ext_intel_power_limits aspect");
       return get_info_impl<UR_DEVICE_INFO_MIN_POWER_LIMIT>();
     }
+    CASE(ext::intel::info::device::luid) {
+      if (!has(aspect::ext_intel_device_info_luid))
+        throw exception(
+            make_error_code(errc::feature_not_supported),
+            "The device does not have the ext_intel_device_info_luid aspect");
+      // TODO: we're essentially memcpy'ing here...
+      static_assert(std::is_same_v<luid_type, std::array<unsigned char, 8>>);
```

Comment:
I think that TODO (where you copy-pasted it from) predates `get_info_impl<UR_<...>>`. Just the C++ type system should be enough here. And I don't think `memcpy`ing part is relevant anymore either.

I suggest just dropping these two lines.

Reply:
Yeah dropping them works fine.